WalletWasabi/Crypto/ZeroKnowledge/ProofSystem.cs

```diff
@@ -194,7 +198,9 @@ public static Statement RangeProofStatement(GroupElement ma, IEnumerable<GroupEl
 			// Ma, proven by showing that the public input is a commitment to 0 (only
 			// Gh term required to represent it). The per-bit witness terms of this
 			// equation are added in the loop below.
-			var bitsTotal = bitCommitments.Select((b, i) => Scalar.Zero.CAddBit((uint)i, 1) * b).Sum();
+			var binaryVector = Enumerable.Range(0, width).Select(i => Scalar.Zero.CAddBit((uint)i, 1)).ToArray();
+			var bitsTotal = new GroupElement(ECMultContext.Instance.MultBatch(binaryVector, b.Select(x => x.Ge).ToArray()).ToGroupElement());
+
 			equations[0, 0] = ma - bitsTotal;
```

Comment:
`ma - bitsTotal` can be computed as part of the batch multiplication, see also next comment

Reply:
What does that mean?

Reply:
if `binaryVector` scalars were all negated (but actually - we can just *not* negate the generator, which will be a bit cleaner too), and prepended with a 1, and the generators were prepended with `ma`, then the final value can be output by `MultBatch` avoiding the negation and regular point addition

Reply:
I got it. I will try it but in advance I think it can improve the performance a bit if we pay the cost for readability/maintainability.